tools/tests/com.liferay.ide.server.core.tests/src/com/liferay/ide/server/core/tests/ServerCoreBase.java

```diff
@@ -130,6 +242,70 @@ public void copyFileToServer( IServer server, String targetFolderLocation, Strin
 
         assertEquals( "Expected the " + file.getName() + " to exist:" + file.getAbsolutePath(), true, file.exists() );
     }
+    
+    protected void deployMavenProject( final IProject project ) throws Exception
+    {
+        
+        assertNotNull( project.getFile( "pom.xml" ) );
+
+        final Job job = new Job( project.getName() + "run deploy" ) //$NON-NLS-1$
+            {
+
+                @Override
+                protected IStatus run( IProgressMonitor monitor )
+                {
+                    try
+                    {
+                        monitor.beginTask( "deploy", 100 );
+
+                        new MavenProjectBuilder( project ).runMavenGoal( project, "package liferay:deploy", monitor );
```

Comment:
The kind of deploy we need to do is the server adapter deploy, like a drag-n-drop.  

Here is snippet of code you can use:

```
ublic static void addProjectToServer(IProject project, IServer server) throws CoreException {
    IModule module  = ServerUtil.getModule(project);
    if (ServerUtil.containsModule(server, module, new NullProgressMonitor())) {
      return;
    }
    IServerWorkingCopy copy = server.createWorkingCopy();
    copy.modifyModules(new IModule[]{module}, new IModule[0], new NullProgressMonitor());
    server = copy.save(true, new NullProgressMonitor());
  }
```